tests/Unit/Utilities/CssConcatenator/RuleBlockTest.php

```diff
@@ -0,0 +1,200 @@
+    }
+
+    /**
+     * @return string[][][]
+     */
+    public function selectorDataProvider(): array
+    {
+        return [
+            'no selectors' => [[]],
+            'one selector' => [['p']],
+            'multiple selectors' => [['p', 'h3']],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string[] $selectors
+     *
+     * @dataProvider selectorDataProvider
+     */
+    public function getCssReturnsCommaSeparatedSelectors(array $selectors)
+    {
+        $subject = new RuleBlock($selectors, '');
+
+        $result = $subject->getCss();
+
+        self::assertSame(\implode(',', $selectors) . '{}', $result);
+    }
+
+    /**
+     * @test
+     */
+    public function getCssCombinesSelectorsAndDeclarationsBlock()
+    {
+        $selectors = ['h1', 'h2'];
+        $declarationsBlock = 'margin-top: 0.5em; padding: 0';
+        $subject = new RuleBlock($selectors, $declarationsBlock);
+
+        $result = $subject->getCss();
+
+        $expectedResult = 'h1,h2{' . $declarationsBlock . '}';
+        self::assertSame($expectedResult, $result);
+    }
+
+    /**
+     * @return string[][][]
+     */
+    public function equivalentSelectorsDataProvider(): array
+    {
+        return [
+            'both empty' => [[], []],
+            'both with the same one selector' => [['p'], ['p']],
+            'both with the same two selectors in the same order' => [['p', 'div'], ['p', 'div']],
+            'both with the same two selectors in a different order' => [['p', 'div'], ['div', 'p']],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string[] $ours
+     * @param string[] $theirs
+     *
+     * @dataProvider equivalentSelectorsDataProvider
+     */
+    public function hasEquivalentSelectorsForEquivalentSelectorsReturnsTrue(array $ours, array $theirs)
+    {
+        $ourRuleBlock = new RuleBlock($ours, 'color: #000;');
+        $theirsRuleBlock = new RuleBlock($theirs, 'color: #000;');
+
+        self::assertTrue($ourRuleBlock->hasEquivalentSelectors($theirsRuleBlock));
+    }
+
+    /**
+     * @return string[][][]
+     */
+    public function notEquivalentSelectorsDataProvider(): array
+    {
+        return [
+            'first empty, second not empty' => [[], ['p']],
+            'first not empty, second empty' => [['p'], []],
+            'different selectors, one each' => [['p'], ['h']],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param string[] $ours
+     * @param string[] $theirs
+     *
+     * @dataProvider notEquivalentSelectorsDataProvider
+     */
+    public function hasEquivalentSelectorsForNotEquivalentSelectorsReturnsFalse(array $ours, array $theirs)
+    {
+        $ourRuleBlock = new RuleBlock($ours, 'color: #000;');
+        $theirsRuleBlock = new RuleBlock($theirs, 'color: #000;');
+
+        self::assertFalse($ourRuleBlock->hasEquivalentSelectors($theirsRuleBlock));
+    }
+
+    // TODO:
+    // - convertDeclarationsBlocks to array internally
```

Comment:
We're currently not doing this because we don't need to for how this is currently used (i.e. putting the applicable uninlinable rules into a `<style>` element).  It could be useful if we're also going to use this class during CSS parsing.  But we might want to alternatively take a look at e.g. [`sabberworm/php-css-parser`](https://github.com/sabberworm/PHP-CSS-Parser) (see https://github.com/MyIntervals/emogrifier/issues/544#issuecomment-526967555) sooner rather than later to avoid duplicating effort...